website/src/views/home.js

```diff
@@ -23,13 +24,24 @@ function Home() {
     const binsInfo = useFetch('/bins');
     const gradeInfo = useFetch(`students/${fetchEmail}/grades`);
     const projectionsInfo = useFetch(`/students/${fetchEmail}/projections`);
+    const navigate = useNavigate();
 
     useEffect(() => {
         if (binsInfo.data && localStorage.getItem('token')) {
             setBinsData(binsInfo.data.map(({ letter, points }) => [points, letter]));
         }
     }, [binsInfo.data]);
 
+
+    // Redirect to error page if any of the fetch requests return an error
+    useEffect(() => {
+        if (binsInfo.error || gradeInfo.error || projectionsInfo.error) {
+            const errorCode = binsInfo.error?.status || gradeInfo.error?.status || projectionsInfo.error?.status;
+            const errorText = binsInfo.error?.message || gradeInfo.error?.message || projectionsInfo.error?.message;
+            navigate('/serviceError', { state: { errorCode, errorText } });
+        }
+    }, [binsInfo.error, gradeInfo.error, projectionsInfo.error, navigate]);
```

Comment:
Objects are compared as effect triggers using their hash meaning every render will trigger this.  You can dereference them in the definitions if you want to pass them in individually, or you could also make a single error state variable that you update with the error of the fetches if if exists.  Then, your useEffect is dependent only on that error and `navigate`.

Reply:
GV-29 is for handling the student not existing in the database. I tried to tackle other errors in the same ticket. I created a new ticket for other redirects for other errors.